syntax = "proto2";

package uss.common_msgs;

import "ultrasonic_detection/common_msgs/error_code.proto";

message Echo {
  //  ultrasonic orientation
  /*
   * front  left->right(1-6):FSL(forword_side_left), FOL, FCL, FCR, FOR, FSR
   * rear left->right(7-12):RSL(backword_side_left), ROL, RCL, RCR, ROR, RSR
   */
  optional uint32 sensor_id = 1;
  //  幅值
  optional uint32 amp = 2;
  //  TOF：发波到收到回波的时间ms
  //  optional uint32 time_duration = 3;
  //  回波宽度：回波强度得到目标回波宽度
  optional uint32 width = 4;
  //  置信度
  optional uint32 confidence = 5;
  //  距离：根据时间算出的目标距离mm
  optional uint32 distance = 6;
  //  错误码
  optional uss.common_msgs.ErrorCode error_code = 7 [default = OK];
  //  测量状态,0表示间接测量，1表示直接测量
  //  optional bool status = 8 [default = 1];
}

message DiagData {
  optional uint32 chip_temp = 1;
  optional uint32 v_sup = 2;
  optional uint32 noise_sum = 3;
  optional uint32 noise_count = 4;  // Burst
  optional uint32 burst_length = 5;
  optional uint32 v_tank1 = 6;
  optional uint32 v_tank2 = 7;  // Ring
  optional uint32 ring_count = 8;
  optional uint32 ring_samples = 9;
  optional uint32 rtm = 10;
}

message NFD {
  optional bool nfd1_flag = 1;
  optional bool nfd2_flag = 2;
  optional uint32 nfd1_echo = 3;
  optional uint32 nfd2_echo = 4;
  optional uint32 nfd3_echo = 5;
  optional uint32 nfd_result = 6;
}

//  algorithm input
message EchoList {
  //  时间戳
  optional uint32 timestamp = 1;
  //  12个探头信号，每个探头输出8个障碍物信息
  repeated Echo echo = 2;
  //  回波数量
  optional uint32 echo_cnt = 3;
  //  诊断信号
  optional DiagData diag_data = 4;
  //  场信号
  optional NFD nfd = 5;
}